packages/electric-db-collection/src/electric.ts

```diff
@@ -158,20 +188,46 @@ export function electricCollectionOptions(
       throw new ExpectedNumberInAwaitTxIdError(typeof txId)
     }
 
+    // First check if the txid is in the seenTxids store
     const hasTxid = seenTxids.state.has(txId)
     if (hasTxid) return true
 
+    // Then check if the txid is in any of the seen snapshots
+    const hasSnapshot = seenSnapshots.state.some((snapshot) =>
```

Comment:
How big are we envisioning this `seenSnapshots` list could become? Worst case we're looping over all snapshots and for each snapshot we check `isVisibleInSnapshot` which in the worst case loop over the snapshot's `xip_list`. So i'm wondering here if this could become a performance issue. If it is, we could as well keep a set of all txIds we've seen in snapshots and then we could just do a set lookup (which is in constant time).